File: app/src/main/java/com/example/desafioandroid/feature/repository/RepositoryViewModelFactory.java
package com.example.desafioandroid.feature.repository;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class RepositoryViewModelFactory implements ViewModelProvider.Factory {

    private Application mApplication;
    private int currentPage;

    public RepositoryViewModelFactory(Application application, int currentPage) {
        this.mApplication = application;
        this.currentPage = currentPage;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new RepositoryViewModel(mApplication, currentPage);
    }
}

File: presentation/src/main/java/com/losingtimeapps/presentation/ui/home/repository/RepositoryViewModelFactory.kt
package com.losingtimeapps.presentation.ui.home.repository

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.losingtimeapps.domain.usercase.GetGitHubRepoUseCase

class RepositoryViewModelFactory(
    private val getGitHubRepoUseCase: GetGitHubRepoUseCase
) : ViewModelProvider.NewInstanceFactory() {

    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return RepositoryViewModel(
            getGitHubRepoUseCase
        ) as T
    }
}